projecttacoma/deqm-test-server-frontend
# Resource count rendering

### Summary

- We added react components to connect to the test server's resourceCounts endpoint, then sort and display each resource and its count as a side menu of the server’s webpage. Each resource is stored as a clickable button which displays the resource’s JSON content in the body of the webpage when clicked.

### New Behavior:
DEQM webpage now displays the resources and their counts as buttons in a side menu.

### Code Changes

- Added components/ResourceCounts.tsx, which fetches all resources and their counts from the backend server. It stores them and their information in an array of buttons displayed on the screen.
- Added components/ResourcePage.tsx, which displays the JSON content of a resource depending on which button was clicked
- Added components/CreateShell.tsx, which creates a shell to separate the header, resource counts, and body of each webpage. Renders the ResourceCounts and ResourcePage components
- Modified pages/_app.tsx to add functionality for Mantine
- Modified pages/index.tsx to render the shell component and an appropriate header
- Added unit_tests/ResourceCounts.test to unit test the new functionality

### Testing Guidance:
- Run the latest test-server (cors was recently added into main, so make sure it's up to date)
- Create a file called `env.local` and add to it: `NEXT_PUBLIC_DEQM_SERVER=http://localhost:3000/4_0_1`
- Run the front end in browser at http://localhost:3001/ and click on any resource button
- Run unit tests
